v1/s012-pyglet/pong.html

```diff
@@ -0,0 +1,616 @@
+                        Musíme tedy zařídit, aby se odrážel od stěn.
+                        Jelikož úhel dopadu se rovná úhlu odrazu,
+                        stačí otočit znaménko y-ové složky rychlosti.
+                        Do funkce <code>obnov_stav(dt)</code> musíme
+                        přidat kontroly na polohu míčku a případně
+                        změnit jeho směr, pokud je moc nízko nebo moc
+                        vysoko.
+<pre>
+def obnov_stav(dt):
+    ...
+    # Odraz micku od sten
+    if pozice_mice[1] < VELIKOST_MICE // 2:
+        rychlost_mice[1] = abs(rychlost_mice[1])
+    if pozice_mice[1] > VYSKA - VELIKOST_MICE // 2:
+        rychlost_mice[1] = -abs(rychlost_mice[1])
+</pre>
+                    </div>
+                    <div>
+                        Teď nám zbývá odraz od pálky, případně resetování
+                        hry, pokud míček padne mimo pálku jednoho hráče a
+                        ten druhý tak získá bod. Opět tedy budeme přidávat
+                        kód do funkce <code>obnov_stav(dt)</code>.
+                    </div>
+                    <div>
+                        Prvním krokem je poznamenání mezí na y-ové ose,
+                        kde se musí míček nacházet, aby byl úspěšně odražen -
+                        to je mezi horním a dolním koncem pálky:
+<pre>
+def obnov_stav(dt):
+    ...
+    palka_min = pozice_mice[1] - VELIKOST_MICE / 2 - DELKA_PALKY / 2
+    palka_max = pozice_mice[1] + VELIKOST_MICE / 2 + DELKA_PALKY / 2
+</pre>
+                        Nyní když míček narazí do pravé nebo levé stěny
+                        se umíme zeptat, zda máme je pálka na správné pozici
+                        a my máme <strong>odrazit</strong> míček nebo zda hráč
+                        prohrál kolo a my máme přičíst jeho soupeři bod a 
+                        <strong>restartovat hru</strong>.
+<pre>
+def obnov_stav(dt):
+    ...
+    # odrazeni vlevo
+    if pozice_mice[0] < TLOUSTKA_PALKY + VELIKOST_MICE / 2:
+        if palka_min < pozice_palek[0] < palka_max:
+            # palka je na spravnem miste, odrazime micek
+            rychlost_mice[0] = abs(rychlost_mice[0])
+        else:
+            # palka je jinde nez ma byt, hrac prohral
+            skore[1] += 1
+            reset()
+
+    # odrazeni vpravo
+    if pozice_mice[0] > SIRKA - (TLOUSTKA_PALKY + VELIKOST_MICE / 2):
+        if palka_min < pozice_palek[1] < palka_max:
+            rychlost_mice[0] = -abs(rychlost_mice[0])
+        else:
+            skore[0] += 1
+            reset()
+</pre>
+                    </div>
+                </section>
+                <section>
+                    <h2>Závěr</h2>
+                    <div>
+                        Hurá, prokousali jsme se k zdárnému konci Pongu.
+                        Máš teď plně funkční interaktivní grafickou
+                        hru zakládající se na reálné předloze:)
+                    </div>
```

Comment:
Původni hra byla trochu složitější, takže možné domácí úkoly:
- Při "odpiknutí" míčku se změní y-ová rychlost podle toho, kde míček pálku zasáhl (uprostřed jednoduchý odraz, u horního okraje letí míček víc nahoru).
- Míček v průběhu hry zrychluje, dokud je na obrazovce. Nový míček pak vždycky začíná s původní rychlostí.
- Nový míček se objeví až pár sekund po tom, co starý vyletí z obrazovky a přičte se skóre